model-optimizer/extensions/ops/loop.py

```diff
@@ -312,14 +313,51 @@ def add_back_edge(loop_node: Node, internal_parameter: Node, internal_result: No
                                      'from_port': 0,
                                      'to_port': 0})
 
+    @staticmethod
+    def parameter_unchanged_after_iteration(loop_node: Node, body_parameter: Node):
+        """
+        Checks if the body Parameter node is connected to some body Result and the data provided to Result is not
+        changed between iterations (so this function call is applicable for Parameters with back edges). The data is
+        considered unchanged if there are only Identity ops in between or Parameter is connected to Result directly.
+
+        :param loop_node: the Loop node to check
+        :param body_parameter: the body Parameter node
+        :return: the result of the check
+        """
+        assert body_parameter.id in loop_node.body
+        assert body_parameter.soft_get('op') == 'Parameter'
+        assert any([attr['to_layer'] == body_parameter.soft_get('internal_layer_id') for attr in loop_node.back_edges])
```

Comment:
It looks a contradiction because you seem to rely on absence of back edges in a caller.